### DISCUSSION TITLE
How to make macro return a value

### DISCUSSION THREAD

**Discussion by imgeopineda** [community/community#121498] [Programming Help]
### Select Topic Area

Question

### Body

I'm trying to call a macro function and return a value. but it keeps returning empty string

here is the macro:

`@Macro("[Select Random Payment Method]")
    fun selectRandomPayment(): String {

        fun <T> List<T>.randomItem(): T {
            val randomIndex = (0 until size).random()
            return get(randomIndex)
        }

        val listPaymentMethods = listOf(
            "BPI - Bank Deposit / Fund Transfer",
            "Chinabank - Bank Deposit / Fund Transfer",
            "RCBC - Bank Deposit / Fund Transfer",
            "UnionBank - Bank Deposit / Fund Transfer",
            "BDO - Pay Bills",
            "Metrobank - Pay Bills"
        )
        return listPaymentMethods.randomItem()
    }`

here is the sample function calling the macro

 val randomPaymentMethod = it.macro("[Select Random Payment Method]").toString()

**Comment by rojanparajuli** [comment]
If you want to call the selectRandomoayment() function and get a random payment method string, you should call it directly rather than trying to invoke it as a macro
val randomPaymentMethod = selectRandomPayment()

**Comment by ebndev** [comment]
Hi @imgeopineda!

Thanks for being a part of the GitHub community! I've moved your post to our Programming Help category, as it may be more likely to get more responses there. Best of luck!
